packages/gasket-plugin-webpack/index.js

```diff
@@ -6,29 +6,37 @@ const { name, devDependencies } = require('./package');
 /**
 * Creates the webpack config
 * @param  {Gasket} gasket The Gasket API
-* @param {Object} webpackConfig Initial webpack config
+* @param {Object} initConfig Initial webpack config
 * @param {Object} data Additional info
 * @returns {Object} Final webpack config
 */
-function initWebpack(gasket, webpackConfig, data) {
-  const { execSync, config } = gasket;
+function initWebpack(gasket, initConfig, data) {
+  const { execSync, execWaterfallSync, config } = gasket;
+
+  const standardPlugins = { plugins: [new WebpackMetricsPlugin({ gasket })] };
 
   const chain = new WebpackChain();
   execSync('webpackChain', chain, data);
 
-  //
-  // Merge defaults with gasket.config webpack.
-  //
-  webpackConfig = webpackMerge.smart(
-    webpackConfig,
-    { plugins: [new WebpackMetricsPlugin({ gasket })] },
-    chain.toConfig(),     // Webpack chain from plugins (partial)
-    config.webpack || {}  // Webpack config from user (partial)
+  const baseConfig = webpackMerge.smart(
+    initConfig,
+    standardPlugins,
+    chain.toConfig(),         // From webpackChain (partial)
+    config.webpack || {}      // From gasket config file (partial)
   );
 
-  const configs = execSync('webpack', webpackConfig, data).filter(Boolean);
+  const configPartials = execSync('webpack', baseConfig, data).filter(Boolean);
+  const mergedConfig = webpackMerge.smart(
+    baseConfig,
+    ...configPartials
+  );
 
-  return webpackMerge.smart(webpackConfig, ...configs);
+  return execWaterfallSync(
+    'webpackMerge',
+    mergedConfig,
+    data,
+    webpackMerge
+  );
```

Comment:
I like what this is doing, but I'm hung up on the naming. This is not really "merging", like the original lifecycle does. I like what you're going for to avoid a breaking major change by introducing a new lifecycle instead of changing up `webpack`. What about a different name, maybe just `webpackConfig`?

Also, the last argument would be most flexible as an object, in case we want to pass along other helpers down the road. For example, it would be really good if we could pass down the imported Webpack version, too.

`execWaterfallSync('webpackConfig', config, data, { webpack, webpackMerge })`

Reply:
Great ideas.

Reply:
@kinetifex: updated, renaming to `webpackConfig` passing an options object.